File: src/specialized/Specialized.Console/Models/Payload/TaskTypeWeight.cs
using Specialized.Console.Models.Enums;

namespace Specialized.Console.Models.Payload;

public record TaskTypeWeight(TaskType Type, int Weight)
{
    public Specialized.Optimizer.Models.Payload.TaskTypeWeight ToProviderModel() =>
        new(Type.ToProviderModel(), Weight);
}

File: src/timefold/src/main/java/com/pso/timefold/domain/TaskTypeWeight.java
package com.pso.timefold.domain;

import com.pso.timefold.domain.enums.TaskType;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
public class TaskTypeWeight {

    private TaskType type;
    private int weight;
}
